internal/api/credentials.go

```diff
@@ -0,0 +1,522 @@
+package api
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/iden3/go-iden3-core/v2/w3c"
+	"github.com/iden3/go-schema-processor/v2/verifiable"
+	"github.com/iden3/iden3comm/v2/packers"
+	"github.com/iden3/iden3comm/v2/protocol"
+
+	"github.com/polygonid/sh-id-platform/internal/common"
+	"github.com/polygonid/sh-id-platform/internal/core/domain"
+	"github.com/polygonid/sh-id-platform/internal/core/ports"
+	"github.com/polygonid/sh-id-platform/internal/core/services"
+	"github.com/polygonid/sh-id-platform/internal/log"
+	"github.com/polygonid/sh-id-platform/internal/repositories"
+	"github.com/polygonid/sh-id-platform/pkg/schema"
+)
+
+// CreateCredential is the creation credential controller. It creates a credential and returns the id
+func (s *Server) CreateCredential(ctx context.Context, request CreateCredentialRequestObject) (CreateCredentialResponseObject, error) {
+	ret, err := s.CreateClaim(ctx, CreateClaimRequestObject(request))
+	if err != nil {
+		return CreateCredential500JSONResponse{N500JSONResponse{Message: err.Error()}}, err
+	}
+	switch ret := ret.(type) {
+	case CreateClaim201JSONResponse:
+		return CreateCredential201JSONResponse(ret), nil
+	case CreateClaim400JSONResponse:
+		return CreateCredential400JSONResponse{N400JSONResponse{Message: ret.Message}}, nil
+	case CreateClaim422JSONResponse:
+		return CreateCredential422JSONResponse{N422JSONResponse{Message: ret.Message}}, nil
+	case CreateClaim500JSONResponse:
+		return CreateCredential500JSONResponse{N500JSONResponse{Message: ret.Message}}, nil
+	default:
+		log.Error(ctx, "unexpected return type", "type", fmt.Sprintf("%T", ret))
+		return CreateCredential500JSONResponse{N500JSONResponse{Message: fmt.Sprintf("unexpected return type: %T", ret)}}, nil
+	}
+}
+
+// CreateClaim is claim creation controller
+// deprecated - Use CreateCredential instead
+func (s *Server) CreateClaim(ctx context.Context, request CreateClaimRequestObject) (CreateClaimResponseObject, error) {
+	did, err := w3c.ParseDID(request.Identifier)
+	if err != nil {
+		return CreateClaim400JSONResponse{N400JSONResponse{Message: err.Error()}}, nil
+	}
+	var expiration *time.Time
+	if request.Body.Expiration != nil {
+		expiration = common.ToPointer(time.Unix(*request.Body.Expiration, 0))
```

Comment:
I'm not sure, but maybe we should use UTC? `common.ToPointer(time.Unix(*request.Body.Expiration, 0).UTC())`
There is an example with UTC: https://pkg.go.dev/time#Unix

Reply:
I don't think so. Using UTC just change the time representation to be in UTC zone, but time is the same. The time.Unix() works in UTC